src/frontends/tensorflow/src/graph_iterator_saved_model.hpp

```diff
@@ -85,68 +85,97 @@ class GraphIteratorSavedModel : public GraphIteratorProto {
         bool res = m_saved_model->ParseFromIstream(&sm_stream);
         FRONT_END_GENERAL_CHECK(res && m_saved_model->meta_graphs_size(), "Saved Model cannot be parsed");
 
+        const ::tensorflow::MetaGraphDef* meta_graph_with_no_tags = nullptr;
+
         for (const auto& meta_graph : m_saved_model->meta_graphs()) {
+            if (meta_graph.meta_info_def().tags_size() == 0) {
+                meta_graph_with_no_tags = &meta_graph;
+            }
+
             if (!meta_graph.has_graph_def()) {
                 continue;
             }
 
-            if (m_saved_model->meta_graphs_size() > 1) {
-                bool tag_found = false;
-                for (const auto& tag : meta_graph.meta_info_def().tags()) {
-                    if (tags.find(tag) != std::string::npos) {
-                        tag_found = true;
-                        break;
-                    }
+            bool tag_found = false;
+            for (const auto& tag : meta_graph.meta_info_def().tags()) {
+                if (tags.find(tag) != std::string::npos) {
```

Comment:
Please double check why `tags` is `std::string`. I though it is a vector of strings

Reply:
Regarding information I found, I'll change this part. We should parse "tags" with ","-delimeter and compare resulting vectors.